README.md

```diff
@@ -110,6 +110,46 @@ end
 
 ```
 
+#### Testing that a ruby method is called with the proper arguments
+
+This assumes you are testing ruby code, not puppet code.
+
+Code under test:
+```ruby
+Puppet::Functions.create_function(:eciton) do
+  dispatch :eciton do
+    param 'String', :some_string
+  end
+
+  def eciton(some_string)
+    Puppet.notice(some_string.reverse)
+  end
+end
+```
+
+To test that the correct string is being set to `notice`...
+
+In your `spec_helper`:
+```ruby
+RSpec.configure do |c|
+  c.mock_with :respec
+end
+```
+
+In your test code:
+```
+require 'spec_helper'
+
+describe 'eciton' do
+  it {
+    expect(Puppet).to receive(:notice).with('gnitset')
```

Comment:
Could also do with a link to the rspec-mock docs: https://relishapp.com/rspec/rspec-mocks/docs